### ISSUE TITLE
False negative: no-unused-vars - spread function's arguments and default values

### ISSUE THREAD

**Issue by l1br3** [eslint/eslint#11712]
**Tell us about your environment**

* **ESLint Version: 5.16**
* **Node Version: 11.2.0**
* **npm Version: 6.4.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
I use standard and babel

**Please show your full configuration:**
I use [standard](https://github.com/standard/standard)
<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function notUsedFalseNegative () {}
function testNotUsed ({
  _notUsedFalseNegative = notUsedFalseNegative
} = {}) {
  notUsedFalseNegative()
}
testNotUsed()
```

```bash
yarn standard MyFile.js
```

**What did you expect to happen?**
The unprefixed `notUsedFalseNegative` version of the function is never used, I expect a warning to tell me I'm not using the good function

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing happened, no warning telling me I'm not using `_notUsedFalseNegative`

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mysticatea** [comment]
Thank you for the report.

Because you are using `standard`. The `standard` doesn't report unused arguments at all. See https://github.com/standard/eslint-config-standard/blob/f7d50e01310866458fd844625cf588d4ee39823b/eslintrc.json#L150.

**Comment by l1br3** [comment]
Thanks for pointing me on the config, I feel it's a really bad standard to
not reporting unused args!

Le jeu. 23 mai 2019 à 13:56, Toru Nagashima <notifications@github.com> a
écrit :

**Comment by platinumazure** [comment]
Closing as the question has been answered and this is not an ESLint issue. I would recommend reporting an issue in the standard repository. Thanks!

**Comment by exogen** [comment]
I believe this should be reopened, or if necessary I can open a new issue.

`args: none` should NOT apply to locals like `_notUsedFalseNegative` in the above snippet, because it is NOT an argument to the function. The function accepts a single argument which is an object. By virtue of it being destructured, that argument is always used.

`_notUsedFalseNegative` is destructured from that object. It is a separate local variable distinguishable from the function argument. I think having `args: none` not report on this situation is dangerous, and indeed like the poster, it has allowed bugs into my program (the reason I'm posting this).

If I have `args: none` set, I would not expect to be warned about this:

```
function (a, b, c) {
  return a;
}
```

But I would indeed expect to be warned about this:

```
function ({ a, b, c }) {
  return a;
}
```

Maybe there could be an additional `args` value that doesn't conflate destructured locals from arguments from the arguments themselves?

**Comment by platinumazure** [comment]
@exogen Please open a new issue, thanks!

**Comment by exogen** [comment]
@platinumazure Done, thanks! https://github.com/eslint/eslint/issues/11879
